WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaPickerHelper.swift

```diff
@@ -58,7 +58,11 @@ class GutenbergMediaPickerHelper: NSObject {
 
         didPickMediaCallback = callback
 
-        let picker = WPNavigationMediaPickerViewController()
+        let mediaPickerOptions = defaultMediaPickerOptions()
+        mediaPickerOptions.filter = filter
+        mediaPickerOptions.allowMultipleSelection = allowMultipleSelection
+
+        let picker = WPNavigationMediaPickerViewController(options: mediaPickerOptions)
```

Comment:
**Disclaimer:** I don't `Swift` often. Most of my historical knowledge is in `Objective-C`.

If we know that `filter` and `allowMultipleSelection` are more frequently use fields on `WPMediaPickerOptions` instances, then should we consider just passing them directly to the `defaultMediaPickerOptions` method with default parameters? 

e.g.,
```
fileprivate func defaultMediaPickerOptions(filter: WPMediaType = [.image],
                                                                          allowMultipleSelection: Bool = false) -> WPMediaPickerOptions {
...
let picker = WPNavigationMediaPickerViewController(options: defaultMediaPickerOptions(filter: filter,
                                                                                                                                                         allowMultipleSelection: allowMultipleSelection))
```

This is just a preference thing as I find I tend to write code more in a declarative style with as few state mutations as possible.

**Aside:** It seems like modern `Swift` still requires the parameters to be named even if the variables passed as arguments to the method match the parameter names identically. I'm not positive though.

Reply:
> declarative style with as few state mutations as possible.
That's a good call @ttahmouch. I'll work to touch this up. 👍

Reply:
This has been changed to a fileprivate extension containing a static function. Let me know your thoughts!

Reply:
> Aside: It seems like modern Swift still requires the parameters to be named even if the variables passed as arguments to the method match the parameter names identically. I'm not positive though.

This is correct AFAIK.

Reply:
The changes read much better to me now. Thanks for addressing, @twstokes . ❤️

The only thing I'm not especially fond of it the use of a static method as that may just make it more difficult to stub if we ever need to mock the options instance for test purposes. I dont think it would be practical to mock with a protocol interface if its static but I may be completely wrong. It also probably doesnt matter too much. I'm not sure if we'd ever need to stage instances for testing.

Reply:
Thanks @ttahmouch - I follow somewhat, but would probably need a concrete example to fully understand. I went with the `static` type method to tie closer to `WPMediaPickerOptions`, but certainly the `fileprivate` visibility would make testing a challenge. Outside of protocols, it may be worthy to consider a `class` type method which would be overridable.

If your proposed `fileprivate func defaultMediaPickerOptions` version is a fix to this problem, could you help me understand better?

Unfortunately none of this is currently being tested as it's riddled with `fileprivate` keywords and types that don't belong to the test target. It would definitely be good to get this in better shape down the road.

Reply:
Ive been reviewing from my phone so it may be tough to show an easy example at the moment, but if I remember Ill try to send one later.

Ive noticed that the test coverage is a pretty common occurrence across the board.

Reply:
Having the options creator method be a method of something that could be injected as a field via a property of the other class would make it easily mockable, e.g., builder pattern or similar. Ideally though, if we were mocking and injecting anything it would probably be a protocol interface wrapped around the Picker; not the Options.

Sorry, phone usage is hard for reviews. :D

Reply:
Ah, I see! Also, props for doing this all on your phone 😆. Yeah, there'd have to be some refactoring all around to make this better with tests.